### PR TITLE
[Issue #154]: Wallet front end list

### PR DESCRIPTION
**Pull request by chedieck**
**Description:**
http://localhost:3000/wallets now shows the real wallets in the database.

**Test plan:**
Should be covered by the tests.
Nonetheless, the reviewer can create paybuttons and then wallets and see if the information is making sense. Check #263 `curl` POST example to see how to create a wallet.

**Remarks:**
- The function to edit the wallets, accessible when clicking in the pencil icon, has not been addressed in this PR and is not expected to work;
- The `default_wallet` is not implemented, so the `wallets` endpoint is returning all wallets as if this property was `false`. This also will be addressed in a next PR.

**Depends on:**
- [x] #266

### CODE REVIEW

File: components/Wallet/WalletCard.tsx [hunk 1]
```diff
@@ -5,50 +5,54 @@ import Image from 'next/image'
...
   return (
     <div className={style.wallet_card}>
       <div className={style.wallet_card_header_ctn}>
         <div className={style.wallet_card_header}>
-          <h4>{walletInfo.name}</h4>
+          <h4>{wallet.name}</h4>
           <div className={style.walletcard_icons}>
-            {walletInfo.xec_balance > 0 && <div><Image src={XECIcon} alt='XEC' /></div>}
-            {walletInfo.bch_balance > 0 && <div><Image src={BCHIcon} alt='BCH' /></div>}
+            {networks.includes(1) && <div><Image src={XECIcon} alt='XEC' /></div>}
```
Comment (ScottMcDermid) [inline components/Wallet/WalletCard.tsx:23]: I'd feel better if we had a more reliable check than `networkId === 1` to indicate the currency of an address.
The client shouldn't have to know network ID's

**Comment by Klakurka** [comment]
We'll want to get the default wallet situation sorted out later.

File: components/Wallet/WalletCard.tsx (hunk 1 above)
Comment (ScottMcDermid) [reply components/Wallet/WalletCard.tsx:23]: This isn't resolved yet. Just wanted to re-open this in case you missed it.

**Comment by Klakurka** [comment]
Conflicts.

**Comment by Klakurka** [comment]
Conflicts

File: components/Wallet/WalletCard.tsx (hunk 1 above)
Comment (chedieck) [reply components/Wallet/WalletCard.tsx]: Indeed I had missed it, thanks. Solved.

File: pages/wallets/index.tsx [hunk 2]
```diff
@@ -81,7 +81,21 @@ class ProtectedPage extends React.Component<WalletsProps, WalletsState> {
...
         <div>
-        {this.state.walletsWithPaymentInfo.sort((a, b) => Number(b.wallet.default_wallet) - Number(a.wallet.default_wallet)).map(walletWithPaymentInfo => {
+        {this.state.walletsWithPaymentInfo.sort((a, b) => {
+          /* Sorts in the following order, from first to last, if they exist:
+           * Default XEC Wallet, Default BCH Wallet, other wallets.
+           */
+          const aNetworkId = Number(a.wallet.userProfile?.isDefaultForNetworkId)
+          const bNetworkId = Number(b.wallet.userProfile?.isDefaultForNetworkId)
+          switch (aNetworkId | bNetworkId) {
+            case 1 | 2:
```
Comment (ScottMcDermid) [inline pages/wallets/index.tsx:91]: I realize this is pulled in from master, but we may as well update these `case` statements to use the network constants

**Review by ScottMcDermid — CHANGES_REQUESTED** [review]

**Review by Klakurka — APPROVED** [review]

**Review by ScottMcDermid — APPROVED** [review]

**Pull request merged**